dl1_data_handler/image_mapper.py

```diff
@@ -72,16 +71,36 @@ def __init__(self,
         self.mask = True if mask_interpolation else False
 
         # Rotate the camera back to the actual orientation. The pixel are moved into a horizontal grid to perform the image_mapper algorithm.
-        self.rot_cam = True if rotate_camera else False
+        self.rotate_camera = True if rotate_back else False
 
-        # Camera geometries, mapping tables and index matrixes initialization
-        self.camera_geometries = {}
+        # Pixel positions, number of pixels, mapping tables and index matrixes initialization
+        self.pixel_positions = {}
+        self.num_pixels = {}
         self.mapping_tables = {}
         self.index_matrixes = {}
 
         for camtype in self.camera_types:
-            # Get a corresponding CameraGeometry
-            self.camera_geometries[camtype] = CameraGeometry.from_name(camtype)
+            # Get a corresponding pixel positions
+            if pixel_positions is None:
+                try:
+                    from ctapipe.instrument.camera import CameraGeometry
+                except ImportError:
+                    raise ImportError("The `ctapipe.instrument.camera` python module is required, if pixel_positions is `None`.")
+                camgeo = CameraGeometry.from_name(camtype)
+                self.num_pixels[camtype] = len(camgeo.pix_id)
+                self.pixel_positions[camtype] = np.column_stack([camgeo.pix_x.value, camgeo.pix_y.value]).T
+                if camtype in ['LSTCam', 'NectarCam', 'MAGICCam']:
+                    rotation_angle = -camgeo.pix_rotation.value * np.pi/180.0
+                    rotation_matrix = np.matrix([[np.cos(rotation_angle), -np.sin(rotation_angle)],
+                                                 [np.sin(rotation_angle), np.cos(rotation_angle)]], dtype=float)
+                    self.pixel_positions[camtype] = np.squeeze(np.asarray(np.dot(rotation_matrix, self.pixel_positions[camtype])))
+                    if self.rotate_camera:
+                        self.rotate_back_angle = 90.0-camgeo.pix_rotation.deg
+            else:
+                self.pixel_positions[camtype] = pixel_positions[camtype]
+                self.num_pixels[camtype] = pixel_positions[camtype].shape[1]
+                if camtype in ['LSTCam', 'NectarCam', 'MAGICCam'] and self.rotate_camera:
+                    self.rotate_back_angle = rotate_back
```

Comment:
Should rotate_back_angle be a dictionary? If there are multiple tel types, only some of which should be rotated, this will set the angle to the value for the last tel type, whatever that happens to be.

Reply:
Right! rotate_back_angle should be a dict.